src/v0/destinations/fb_custom_audience/transform.js

```diff
@@ -292,13 +294,16 @@ const processEvent = (message, destination) => {
   if (message.type.toLowerCase() !== 'audiencelist') {
     throw new InstrumentationError(` ${message.type} call is not supported `);
   }
-  const operationAudienceId = audienceId;
-
+  let operationAudienceId = audienceId;
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  if (!operationAudienceId && mappedToDestination) {
```

Comment:
audienceId is a required field in the dashboard, so this logic will never be true.

Reply:
no fore source of type warehouse `audienceId` will not be taken from dashboard. Thus this logic is needed please check test payloads.